moped-editor/src/views/projects/projectView/DropDownSelectComponent.js

```diff
@@ -1,78 +1,72 @@
-import React from "react";
+import React, { useMemo, useEffect } from "react";
 import { handleKeyEvent } from "../../../utils/materialTableHelpers";
-import {
-  Select,
-  MenuItem,
-  FormControl,
-  FormHelperText,
-} from "@material-ui/core";
-import { phaseNameLookup } from "src/utils/timelineTableHelpers";
+import { Select, MenuItem, FormControl } from "@material-ui/core";
+
+// hook which keeps available subphase options in sync
+const useSubphaseOptions = ({ phases, phaseId }) =>
+  useMemo(() => {
+    if (!phaseId || !phases) return [];
+    // Find this phase's related subphases
+    const allowedSubphases = phases.find(
+      (item) => item?.phase_id === Number(phaseId)
+    )?.moped_subphases;
+    // Shape subphases for menu options
+    return allowedSubphases.map((subphase) => ({
+      value: subphase.subphase_id,
+      label: subphase.subphase_name,
+    }));
+  }, [phases, phaseId]);
 
 /**
  * DropDownSelectComponent - Renders a drop down menu for MaterialTable
  * @param {object} props - Values passed through Material Table `editComponent`
- * @param {string} name - Field name
  * @return {JSX.Element}
  * @constructor
  */
-const DropDownSelectComponent = (props) => {
-  // If the component name is phase_name, then use phaseNameLookup values, otherwise set as null
-  let lookupValues =
-    props.name === "phase_name" ? phaseNameLookup(props.data) : null;
+const DropDownSelectComponent = ({ name, data, rowData, value, onChange }) => {
+  const phaseId = rowData?.moped_phase?.phase_id;
+  const phases = data.moped_phases;
 
-  // If lookup values is null, then it is a sub-phase list we need to generate
-  if (lookupValues === null) {
-    // First retrieve the sub-phase id's from moped_phases for that specific row
-    const allowedSubphases = props.data.moped_phases.find(
-      (item) => item?.phase_id === Number(props.rowData?.phase_id ?? 0)
-    )?.moped_subphases;
+  const subphaseOptions = useSubphaseOptions({
+    phases,
+    phaseId,
+  });
 
-    // If there are no subphases, hide the drop-down.
-    if (!allowedSubphases || allowedSubphases.length === 0) {
-      return null;
+  useEffect(() => {
+    // this effect ensures the subphase value is reset when the row's
+    // phase changes
+    if (value && !subphaseOptions.find((option) => option.value === value)) {
+      onChange("");
     }
+  }, [value, subphaseOptions, phaseId]);
```

Comment:
this fixes a production bug that caused the previous subphase selection to persist and be committed to the DB if the user changed the row's phase after selecting a subphase.